xarray/core/parallel.py

```diff
@@ -0,0 +1,86 @@
+try:
+    import dask
+    import dask.array
+    from dask.highlevelgraph import HighLevelGraph
+
+except ImportError:
+    pass
+
+from .dataarray import DataArray
+
+
+def get_chunk_slices(dataset):
+    chunk_slices = {}
+    for dim, chunks in dataset.chunks.items():
+        slices = []
+        start = 0
+        for chunk in chunks:
+            stop = start + chunk
+            slices.append(slice(start, stop))
+            start = stop
+        chunk_slices[dim] = slices
+
+    return chunk_slices
+
+
+def map_blocks(func, darray):
```

Comment:
We might consider accepting `*args, **kwargs` here, at least in the simple case when they aren't other Dask/Xarray collections.  That would allow for simple things like ...

```python
darray.map_blocks(operator.add, 10)
```

So maybe something like ...

```python
for arg in args + tuple(kwargs.values()):
    if dask.base.is_dask_collection(arg):
        raise NotImplementedError(...)
```

And then use the task`(_wrapper, func, (DataArray, key, ...), args, kwargs)`

Reply:
+1 for ``*args, **kwargs``